adaptivefiltering/widgets.py

```diff
@@ -335,3 +334,58 @@ def _setter(_d):
             setter=_setter,
             widgets=[widget],
         )
+
+
+def upload_files(directory=None, filetype=""):
+    """
+    Create a widget to upload and store files over the jupyter interface.
+    This function will be called by the different load functions.
+
+    :param directory:
+        The directory on the server where the files will be saved.
+        This will be set by the function calling upload_files and be representive of the different upload types.
+        eg. Pipelines will be saved in a differend directory than datasets or segmentations.
+    :type directory: string
+    :param filetype:
+        Set the filetype filter for the upload widget
+    :type filetype: string
+
+    :return: The name(s) of the uploaded file(s)
+
+    """
+    # this needs to be loaded here to avoid circular imports
+    from adaptivefiltering.apps import block_until_button_click
+
+    if directory is None:
+        print("Uploaded files will be saved in the current working directory.")
+    if not os.path.isdir(directory):
+        print("The directory: " + directory + "does not exist and will be created.")
+        os.mkdir(directory)
+
+    confirm_button = ipywidgets.Button(
+        description="Confirm upload",
+        disabled=False,
+        button_style="",  # 'success', 'info', 'warning', 'danger' or ''
+        tooltip="Confirm upload",
+        icon="check",  # (FontAwesome names without the `fa-` prefix)
+    )
+    upload = ipywidgets.FileUpload(
+        accept=filetype,  # Accepted file extension e.g. '.txt', '.pdf', 'image/*', 'image/*,.pdf'
+        multiple=True,  # True to accept multiple files upload else False
+    )
+
+    app = ipywidgets.AppLayout(
+        header=None,
+        footer=ipywidgets.Box([upload, confirm_button]),
+        pane_widths=[1, 0, 2],
+    )
+    display(app)
+    block_until_button_click(confirm_button)
+    app.layout.display = "none"
+    uploaded_data = upload.value
+    filenames = []
+    for filename, uploaded_file in uploaded_data.items():
+        filenames.append(filename)
+        with open(directory + "/" + filename, "wb") as fp:
+            fp.write(uploaded_file["content"])
+    return ["./" + directory + "/" + name for name in filenames]
```

Comment:
Filename/Path concatenations should be done with `os.path.join` to ensure their portability across platforms.